lib/pxlogserviceclient.js

```diff
@@ -0,0 +1,37 @@
+const { EXTERNAL_LOGGER_SERVICE_PATH } = require('./utils/constants');
+
+class LogServiceClient {
+    constructor(config, pxClient) {
+        this.config = config;
+        this.appId = config.PX_APP_ID;
+        this.pxClient = pxClient;
+    }
+
+    sendLogs(pxCtx, logs, req) {
+        try {
+            const enrichedLogs = logs.map((log) => this.enrichLogRecord(pxCtx, log, req));
+            this.postLogs(enrichedLogs);
+        } catch (e) {
+            this.config.logger.error(`unable to send logs: + ${e}`);
+        }
+    }
+
+    enrichLogRecord(pxCtx, logs, req) {
+        const logMetadata = {
+            container: 'enforcer',
+            appID: this.appId,
+            method: pxCtx ? pxCtx.httpMethod : req.method || '',
+            host: pxCtx ? pxCtx.hostname : req.hostname || req.get('host'),
+            path: pxCtx ? pxCtx.uri : req.originalUrl || '/',
+            requestId: pxCtx ? pxCtx.requestId : ''
+        };
+
+        return { ... logMetadata, ...logs };
+    }
+
+    postLogs(enrichLogs) {
+        this.pxClient.callServer(enrichLogs, EXTERNAL_LOGGER_SERVICE_PATH, {}, this.config);
+    }
+}
+
+module.exports = LogServiceClient;
```

Comment:
I always prefer to do named exports rather than default exports to prevent typos and make future refactoring easier.

```js
module.exports = { LogServiceClient };
```

Reply:
done